src/h3/stream.rs

```diff
@@ -0,0 +1,516 @@
+// Copyright (C) 2019, Cloudflare, Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use super::frame::Frame;
+use super::Error;
+use super::Result;
+
+use crate::octets;
+
+pub const HTTP3_CONTROL_STREAM_TYPE_ID: u8 = 0x43;
+pub const HTTP3_PUSH_STREAM_TYPE_ID: u8 = 0x50;
+pub const QPACK_ENCODER_STREAM_TYPE_ID: u8 = 0x48;
+pub const QPACK_DECODER_STREAM_TYPE_ID: u8 = 0x68;
+
+#[derive(Clone, Copy, PartialEq)]
+pub enum StreamType {
+    Control,
+    Request,
+    Push,
+    QpackEncoder,
+    QpackDecoder,
+    // Grease, // TODO: enable GREASE streams
+}
+
+#[derive(Clone, Copy, PartialEq)]
+pub enum StreamState {
+    StreamTypeLen,
+    StreamType,
+    FrameTypeLen,
+    FrameType,
+    FramePayloadLenLen,
+    FramePayloadLen,
+    FramePayload,
+    PushIdLen,
+    PushId,
+    QpackInstruction,
+    Invalid,
+    Done,
+}
+
+impl std::fmt::Debug for StreamState {
```

Comment:
This whole thing seems to be equivalent to just deriving `Debug` for `StreamState` (i.e. can just add `Debug` to the `#[derive()]` list, note that it should be in alphabetical order for style).